library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -51,4 +54,33 @@ object JSConverters {
       arr: scala.Array[T]): JSRichGenTraversableOnce[T] =
     new JSRichGenTraversableOnce(arr)
 
+  implicit class JSRichFuture[A](val self: Future[A]) {
+    /** Converts the Future to a JavaScript [[Promise]].
+     *
+     *  Attention! The nature of the [[Promise]] class, from the ECMAScript
+     *  specification, makes this method inherently un-typeable, because it is
+     *  not type parametric.
+     *
+     *  The signature of the `toJSPromise` method is only valid
+     *  <i>provided that</i> the values of `A` do not have a `then` method.
+     */
+    def toJSPromise: Promise[A] = {
```

Comment:
I'm wondering if we should implicit foo to actually resolve `Thenable`s properly here. Of course that doesn't give you any guarantee, since your object might have a `then ` method, but at least you can do something if you know:

```scala
val f0: Future[js.Dynamic] = ???
// Assume I know `f0`s result has a `then` method
val f1 = f0.asInstanceOf[Future[Thenable[js.Dynamic]]
f1.toJSPromise // sound
```